benchmarks/renderImageItem.py

```diff
@@ -22,117 +23,183 @@ def renderQImage(*args, **kwargs):
+    unit = "seconds"
+    param_names = ["size", "acceleration", "use_levels", "dtype", "channels", "lut_length"]
+    params = Parameters(
+        [
+            # (256, 256),
+            # (512, 512),
+            # (1024, 1024),
+            # (2048, 2048),
+            # (3072, 3072),
+            (4096, 4096)
+        ],                                      # size
+        ["numpy", "numba", "cupy"],             # acceleration
+        [True, False],                          # use_levels
+        ['uint8', 'uint16', 'float32'],         # dtype
+        [1, 3],                                 # channels
+        ['uint8', 'uint16', None]               # lut_length
+    )
     def __init__(self):
-        super(_TimeSuite, self).__init__()
-        self.size = None
-        self.float_data = None
-        self.uint8_data = None
-        self.uint8_lut = None
-        self.uint16_data = None
-        self.uint16_lut = None
-        self.cupy_uint16_lut = None
-        self.cupy_uint8_lut = None
-
-    def setup(self):
-        size = (self.size, self.size)
-        self.float_data, self.uint16_data, self.uint8_data, self.uint16_lut, self.uint8_lut = self._create_data(
-            size, np
-        )
-        if numba is not None:
+        self.data = np.empty((), dtype=np.uint8)
+        self.lut = np.empty((), dtype=np.ubyte)
+        self.levels = None
+
+    def teardown(self, *args, **kwargs):
+        # toggle options off
+        pg.setConfigOption("useNumba", False)
+        pg.setConfigOption("useCupy", False)
+
+    def setup_cache(self) -> dict:
+        accelerations = [np]
+        if cp is not None:
+            accelerations.append(cp)
+
+        cache = {}
+
+        for xp in accelerations:
+            cache[xp.__name__] = {"lut": {}, "data": {}}
+
+            # handle lut caching
+            c_map = xp.array([[-500.0, 255.0], [-255.0, 255.0], [0.0, 500.0]])
+            for lut_length in self.params.lut_lengths:
+                if lut_length is None:
+                    continue
+                bits = xp.dtype(lut_length).itemsize * 8
+                # create the LUT
+                lut = xp.zeros((2 ** bits, 4), dtype="ubyte")
+                for i in range(3):
+                    lut[:, i] = xp.clip(xp.linspace(c_map[i][0], c_map[i][1], 2 ** bits), 0, 255)
+                lut[:, -1] = 255
+                cache[xp.__name__]["lut"][lut_length] = lut
+
+            # handle data caching
+            for dtype in self.params.dtypes:
+                cache[xp.__name__]["data"][dtype] = {}
+                for channels in self.params.channels:
+                    cache[xp.__name__]["data"][dtype][channels] = {}
+
+                    for size in self.params.sizes:
+                        size_with_channels = (size[0], size[1], channels) if channels != 1 else size
+                        if xp.dtype(dtype) in (xp.float32, xp.float64):
+                            data = xp.random.default_rng().standard_normal(
```

Comment:
Calling `default_rng()` instantiates a new Generator object. Was that intended or could a single Generator object be used?

Reply:
I just went w/ the example in the docs; you're right, should probably just use a single generator object